runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/graph/ExecutableStage.java

```diff
@@ -139,14 +151,27 @@ default PTransform toPTransform() {
    */
   static ExecutableStage fromPayload(ExecutableStagePayload payload, Components components) {
     Environment environment = payload.getEnvironment();
-    PCollectionNode input = PipelineNode.pCollection(payload.getInput(),
-        components.getPcollectionsOrThrow(payload.getInput()));
-    List<PTransformNode> transforms = payload.getTransformsList().stream()
-        .map(id -> PipelineNode.pTransform(id, components.getTransformsOrThrow(id)))
-        .collect(Collectors.toList());
-    List<PCollectionNode> outputs = payload.getOutputsList().stream()
-        .map(id -> PipelineNode.pCollection(id, components.getPcollectionsOrThrow(id)))
-        .collect(Collectors.toList());
-    return ImmutableExecutableStage.of(environment, input, transforms, outputs);
+    PCollectionNode input =
+        PipelineNode.pCollection(
+            payload.getInput(), components.getPcollectionsOrThrow(payload.getInput()));
```

Comment:
At this pointm it may be worth wrapping the getXOrThrow calls to actually return a legible error message since the generated code does not print the input ids.

Reply:
https://issues.apache.org/jira/browse/BEAM-3899, leaving for follow-ups